doc/rbd/rbd-snapshot.rst

```diff
@@ -206,107 +232,131 @@ performed these steps, you can begin cloning the snapshot.
+   updated just like the base image.
+
+#. **Template Pool:** One way to use block device layering is to create a pool
+   that contains both (1) master images that act as templates and (2) snapshots
+   of those templates. You may then extend read-only privileges to users so
+   that they may clone the snapshots without the ability to write or execute
+   within the pool.
+
+#. **Image Migration/Recovery:** One way to use block device layering is to
+   migrate or recover data from one pool into another pool.
 
 Protecting a Snapshot
 ---------------------
 
-Clones access the parent snapshots. All clones would break if a user inadvertently 
-deleted the parent snapshot. To prevent data loss, you **MUST** protect the
-snapshot before you can clone it. ::
+Clones access the parent snapshots. All clones would break if a user
+inadvertently deleted the parent snapshot. To prevent data loss, you **MUST**
+protect the snapshot before you can clone it:
+
+.. prompt:: bash $
 
-	rbd snap protect {pool-name}/{image-name}@{snapshot-name}
+   rbd snap protect {pool-name}/{image-name}@{snapshot-name}
 
-For example::
+For example:
 
-	rbd snap protect rbd/my-image@my-snapshot
+.. prompt:: bash $
+
+   rbd snap protect rbd/my-image@my-snapshot
 
 .. note:: You cannot delete a protected snapshot.
 
 Cloning a Snapshot
 ------------------
 
-To clone a snapshot, specify you need to specify the parent pool, image and
-snapshot; and, the child pool and image name. You must protect the snapshot
-before  you can clone it. ::
+To clone a snapshot, specify the parent pool, the image, and the snapshot; and
+also the child pool together with the image name. You must protect the snapshot
+before you can clone it:
+
+.. prompt:: bash $
 
-	rbd clone {pool-name}/{parent-image}@{snap-name} {pool-name}/{child-image-name}
+   rbd clone {pool-name}/{parent-image}@{snap-name} {pool-name}/{child-image-name}
 	
-For example:: 
+For example:
 
-	rbd clone rbd/my-image@my-snapshot rbd/new-image
+.. prompt:: bash $
+
+   rbd clone rbd/my-image@my-snapshot rbd/new-image
 	
-.. note:: You may clone a snapshot from one pool to an image in another pool. For example, 
-   you may maintain read-only images and snapshots as templates in one pool, and writeable
-   clones in another pool.
+
+.. note:: You may clone a snapshot from one pool to an image in another pool.
+   For example, you may maintain read-only images and snapshots as templates in
+   one pool, and writeable clones in another pool.
 
 Unprotecting a Snapshot
 -----------------------
 
-Before you can delete a snapshot, you must unprotect it first. Additionally,
+Before you can delete a snapshot, you must first unprotect it. Additionally,
 you may *NOT* delete snapshots that have references from clones. You must
-flatten each clone of a snapshot, before you can delete the snapshot. :: 
+flatten each clone of a snapshot before you can delete the snapshot:
```

Comment:
"must flatten or delete each clone of a snapshot before you can unprotect the snapshot"

Reply:
Done.